src/components/Staff/Checkin/QR.tsx

```diff
@@ -0,0 +1,99 @@
+import type { CheckInDataDTO, UserShowedData } from "@/types/staff";
+import { Result } from "@zxing/library";
+import { motion } from "framer-motion";
+import { useEffect, useState } from "react";
+import { QrReader } from "react-qr-reader";
+import Failed from "./Failed";
+import User from "./User";
```

Comment:
I recommend organizing imports by:

1. External libraries
2. Internal components
3. Images, data, and types